[TLVB-145] 사업자 가게 조회 API 수정

dia312 opened the pull request:
## 👩‍💻 요구 사항과 구현 내용
- 사업자 가게 조회 API(API - 20) 수정 
- 최초 사업자 전환 시, 가게를 등록할 수 있으므로 Market 테이블의 User 필드에 대해 unique 설정
- 사업자 가게 조회 페이지에 있는 count 정보(review, like, event)에 대해 @Formula 어노테이션 적용

## ✅ 피드백 반영사항

## 🤔 PR 포인트 & 궁금한 점

## 관련 이슈 
TLVB-145

dia312 committed:
feat: 리뷰 카운트 증가 기능 구현

dia312 committed:
fix: 리뷰 카운트 증가 기능 재거

dia312 committed:
feat: 사업자 가게 조회 API 수정

nasaoreo commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/Market.java:
```diff
@@ -51,6 +65,18 @@ public void minusOneFavorite() {
     this.favoriteCount--;
   }
 
+  public int getLikeCount() {
+    return likeCount;
+  }
+
+  public int getReviewCount() {
+    return reviewCount;
+  }
+
+  public int getEventCount() {
+    return eventCount;
+  }
```
앗 에디 지금 저랑 작업하고 있는거 또 겹쳤네요ㅋㅋㅋ 저도 여기 부분 하고 있었는데ㅋㅋㅋㅋ
다른 엔티티와 일관성 있게 Lombok Getter 쓰는 것 어떤가용? 혹시 Formula를 사용해서 get 함수가 따로 필요한가욤?~

dia312 replied:
게터잇는걸 깜빡햇네용 ㅋㅋㅋㅋㅋㅋ

nasaoreo commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/dto/MarketReadResponse.java:
```diff
@@ -2,10 +2,9 @@
 @Builder
 public class MarketReadResponse {
-    private Page<SimpleMarket> simpleMarkets;
+    private MyMarket myMarket;
```
이전에는 일급 컬렉션의 형태로 만들기 위해 MarketReadResponse가 simpleMarket 리스트를 가지고 있었는데,
지금처럼 단 하나의 Market 정보만 가지고 있다면 이 클래스에 바로 MyMarket 클래스의 필드를 넣어도 되지 않을까요~?

dia312 replied:
조아용ㅎㅎ 수정하구 반영하겟습니다!

nasaoreo reviewed (approved):
에디😸  동시에 작업하고 있는 부분이 많네요ㅋㅋㅋ 수고하셨습니당~! 👍

dia312 committed:
fix: 코드리뷰 피드백 반영
- MarketControlleTest 삭제
- api/v1/markets(get) Response 변경
- market의 reviewCount, likeCount, eventCount getter 중복 삭제

dia312 committed:
feat: /markets(get) response 데이터에 name 추가

dia312 committed:
fix: 현재 develop 브랜치와 병합

dia312 committed:
fix: MarketServiceTest 사용되지 않는 코드 삭제

Heejeong01110 reviewed (approved):
고생하셨습니다~!!!

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/Market.java:
```diff
@@ -43,6 +44,19 @@
   @Column(nullable = false)
   private int favoriteCount;
 
+  @Formula("(select count(1) from Event_Like el "
+      + "join event e on e.id = el.event_id "
+      + "where e.market_id = id)")
+  private int likeCount;
+
+  @Formula("(select count(1) from Review r "
+      + "join event e on e.id = r.event_id "
+      + "where e.market_id = id)")
+  private int reviewCount;
+
+  @Formula("(select count(1) from Event e where e.market_id = id)")
+  private int eventCount;
```
1이라는것보다는 컬럼명을 명시하는걸 추천드립니다.
아울러 예약어와 컬럼명 및 테이블명을 구분하는게 더 좋지 않을까요?
예를들면 예약어를 대문자로 쓰는거죠 :)

dia312 replied:
신경쓰지 못한 것 같아요 ㅠㅠ 반영하겠습니다 !! 감사합니다 😄

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/market/repository/MarketRepository.java:
```diff
@@ -11,5 +12,5 @@
 @Repository
 public interface MarketRepository extends JpaRepository<Market, Long> {
-    Page<Market> findByUser(User user, Pageable pageable);
+    Optional<Market> findByUser(User user);
```
한명의 유저는 한개의 가게만 가질수 있나보군요!?

ksy90101 replied:
그렇다면 유니크 인덱스를 걸면 어떨까요?

dia312 replied:
넵 맞습니다!! 그래서 Market.java 의 User 컬럼에 unique 설정을 했습니다ㅎㅎ
  ```Market.java
@ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(referencedColumnName = "id", unique = true)
  private User user;

```

ksy90101 replied:
아 제가 잘못생각했네요.
근데 지금처럼 하셨으면 사실 되기도 했네요 👍
